test/chrome.test.js

```diff
@@ -0,0 +1,59 @@
+import { getCurrentTabUrl } from '../src/api/chrome'
+
+let sendResponseSpy
+beforeEach(() => {
+  const originalImpl = global.chrome.runtime.onMessage.addListener // prevent infinite-loop
+  jest
+    .spyOn(global.chrome.runtime.onMessage, 'addListener')
+    .mockImplementation(listener => {
+      const mockSpiedListener = jest.fn(listener)
+      originalImpl(mockSpiedListener)
+    })
+  require('../src/background')
+  chrome.windows.getCurrent.mockImplementation(async () => {
+    return [{ id: 1 }]
+  })
+  sendResponseSpy = jest.fn()
+  chrome.runtime.sendMessage.mockImplementation(action => {
+    return new Promise(resolve => {
+      sendResponseSpy.mockImplementation((...args) => {
+        resolve(...args)
+      })
+      chrome.runtime.onMessage.callListeners(action, {}, sendResponseSpy)
+    })
+  })
+})
+
+afterEach(() => {
+  chrome.runtime.onMessage.clearListeners()
+  jest.resetAllMocks()
+  jest.restoreAllMocks()
+  jest.resetModules()
```

Comment:
this looks very hackish. also, cleanups are better done in before functions because when a test start, you can't really assume that whatever test ran before took care of cleaning up

Reply:
I'd say that it depends on the approach. I used to do clean up in the `after` because I expect that every module does proper clean up :D. In order to make these tests bullet proof I'd extract these calls and call it  in `after` and in `before` each  :). wdyt?